### ISSUE TITLE
Dynamic linker error when executing `ApolloClient.fetch`

### ISSUE THREAD

**Issue by gbrixey** [apollographql/apollo-ios#640]
After updating to version 0.11.0 and 0.11.1 via Carthage, I started to experience a crash at runtime whenever executing `ApolloClient.fetch` or `ApolloClient.perform`. I downgraded to 0.10.1 and the issue stopped occurring. I can't figure out if there is something on my side that should be changed to fix this issue. Any ideas?

Crashing with the following message when using `ApolloClient.fetch`
```
dyld: lazy symbol binding failed: Symbol not found: _$s6Apollo0A6ClientC5fetch5query11cachePolicy5queue13resultHandlerAA11Cancellable_px_AA05CacheF0OSo012OS_dispatch_G0CyAA13GraphQLResultVy4DataQzGSg_s5Error_pSgtcSgtAA0N7QLQueryRzlF
Referenced from: /Users/glenb/Library/Developer/CoreSimulator/Devices/48D18C6E-4D40-4976-BB33-D89E70C78E5A/data/Containers/Bundle/Application/BCC304E0-8E7A-4F7F-B87D-0B6A4219F16A/...
  Expected in: /Users/glenb/Library/Developer/CoreSimulator/Devices/48D18C6E-4D40-4976-BB33-D89E70C78E5A/data/Containers/Bundle/Application/BCC304E0-8E7A-4F7F-B87D-0B6A4219F16A/.../Apollo.framework/Apollo
```

When using `ApolloClient.perform`:
```
dyld: lazy symbol binding failed: Symbol not found: _$s6Apollo0A6ClientC7perform8mutation5queue13resultHandlerAA11Cancellable_px_So012OS_dispatch_E0CyAA13GraphQLResultVy4DataQzGSg_s5Error_pSgtcSgtAA0K10QLMutationRzlF
  Referenced from: /Users/glenb/Library/Developer/CoreSimulator/Devices/48D18C6E-4D40-4976-BB33-D89E70C78E5A/data/Containers/Bundle/Application/F8D89A9A-C1AF-4491-AA54-ECEF4C713250/...
  Expected in: /Users/glenb/Library/Developer/CoreSimulator/Devices/48D18C6E-4D40-4976-BB33-D89E70C78E5A/data/Containers/Bundle/Application/F8D89A9A-C1AF-4491-AA54-ECEF4C713250/.../Apollo.framework/Apollo
```

**Comment by designatednerd** [comment]
Seems like [on PromiseKit a similar issue was solved by nuking the Carthage folder and building again](https://github.com/mxcl/PromiseKit/issues/1006) - can you give that a shot and let me know if it works?

**Comment by gbrixey** [comment]
I tried deleting the Carthage folder and running `carthage bootstrap` again, but I am experiencing the same issue.

**Comment by designatednerd** [comment]
Weird! We haven't had any other reports of this, so it does seem like there's a build artifact that's borked somewhere in your system. I assume you've done a `Clean Build Folder` at some point? 

If you haven't already tried it, try closing Xcode, deleting your `DerivedData` folder (usually in `/Users/[your user]/Library/Developer/Xcode/DerivedData/`), doing the delete / reinstall Carthage dance, then opening Xcode and rebuilding your project. This will clear out all previous build artifacts on your system, and force everything to rebuild from scratch.

If that doesn't work, we'll have to do some deeper digging.

**Comment by gbrixey** [comment]
OK, I have deleted and rebuilt pretty much everything, and the crash has stopped happening. Something I neglected to mention before is that the app I am working on receives the Apollo dependency through another internal framework. I also rebuilt that internal framework and all its Carthage dependencies, which I had not tried before, and created a new binary of the internal framework. I'm not sure, but I think that was the step that resolved the runtime error.

Thanks for your help / sorry for bothering you @designatednerd 🙇
